packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -115,13 +131,26 @@ function getComponentStateValue(
     | CheckboxElement
     | SelectorElement
     | AccountSelectorElement,
+  getAccount: GetAccount,
 ) {
   switch (element.type) {
     case 'Checkbox':
       return element.props.checked;
 
-    case 'AccountSelector':
-      return element.props.selectedAddress;
+    case 'AccountSelector': {
+      if (!element.props.selectedAccount) {
+        return undefined;
+      }
+
+      const account = getAccount(element.props.selectedAccount);
+
+      assert(
+        account,
+        `Account with ID ${element.props.selectedAccount} not found.`,
+      );
```

Comment:
Maybe just return `undefined` or `null` in that case?

Reply:
I prefer to throw because we then don't know which account to select since it always need to have an account selected